pkg/interceptors/server/server.go

```diff
@@ -236,3 +240,46 @@ func (is *Server) UpdateCRDWithCaCert(ctx context.Context, triggersV1Alpha1 trig
 	}
 	return nil
 }
+
+func checkCertValidity(serverCert, caCert []byte, logger *zap.SugaredLogger) {
+	ticker := time.NewTicker(time.Minute)
+	quit := make(chan struct{})
+	var (
+		cert *x509.Certificate
+		err  error
+	)
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				// Check the expiration date of the certificate to see if it needs to be updated
+				roots := x509.NewCertPool()
+				ok := roots.AppendCertsFromPEM(caCert)
+				if !ok {
+					logger.Error("failed to parse root certificate")
+				}
+				block, _ := pem.Decode(serverCert)
+				if block == nil {
+					logger.Error("failed to parse certificate PEM")
+				} else {
+					cert, err = x509.ParseCertificate(block.Bytes)
+					if err != nil {
+						logger.Errorf("failed to parse certificate: %v", err.Error())
+					}
+				}
+
+				opts := x509.VerifyOptions{
+					Roots: roots,
+				}
+
+				if _, err := cert.Verify(opts); err != nil {
+					logger.Errorf("failed to verify certificate: %v", err.Error())
```

Comment:
So, if we fail to validate the certificate - what do we do next? It seems like all we are doing is logging an error, should this be an indication to update the cert?

Reply:
Yes its an indication to update the cert